edge/services/cpu_percent/Makefile

```diff
@@ -1,47 +1,62 @@
-# Transform the machine arch into some standard values: "arm", "arm64", or "amd64"
-SYSTEM_ARCH := $(shell uname -m | sed -e 's/aarch64.*/arm64/' -e 's/x86_64.*/amd64/' -e 's/armv.*/arm/')
+# This imports the variables from horizon/hzn.json. You can ignore these lines, but do not remove them.
+-include horizon/.hzn.json.tmp.mk
 
 # To build for an arch different from the current system, set this env var to 1 of the values in the comment above
-export ARCH ?= $(SYSTEM_ARCH)
+export ARCH ?= $(shell hzn architecture)
 
 # These variables can be overridden from the environment
-export CPU_NAME ?= cpu
-export CPU_VERSION ?= 1.2.2
-DOCKER_NAME ?= $(ARCH)_$(CPU_NAME)
-export DOCKER_HUB_ID ?= openhorizon
-export MYDOMAIN ?= github.com.open-horizon.examples
-PORT_NUM ?= 8347
-IMAGE_PATH ?= $(DOCKER_HUB_ID)/$(DOCKER_NAME):$(CPU_VERSION)
-
-default: all
-
-all: build run check
+PORT_NUM ?= 80
 
 build:
-	docker build -t $(IMAGE_PATH) -f ./Dockerfile.$(ARCH) .
-
-run:
-	-docker rm -f $(DOCKER_NAME) 2> /dev/null || :
-	docker run -d --name $(DOCKER_NAME) --publish=$(PORT_NUM):$(PORT_NUM) --volume `pwd`:/outside $(IMAGE_PATH)
-
-check:
+	docker build -t $(DOCKER_IMAGE_BASE)_$(ARCH):$(SERVICE_VERSION) -f ./Dockerfile.$(ARCH) .
+
+build-all-arches:
+	ARCH=amd64 $(MAKE) build
+	ARCH=arm $(MAKE) build
+	ARCH=arm64 $(MAKE) build
+
+# In travis can only read syslog via sudo
+ifeq ($(TRAVIS),true)
+SYSLOG_SUDO = sudo
+else
+SYSLOG_SUDO =
+endif
+# Soon we can use 'hzn service log ...' for all platforms
+ifeq ($(shell uname -s),Darwin)
+GET_SVC_OUTPUT = docker logs `docker ps -q --filter name=cpu`
+else
+GET_SVC_OUTPUT = $(SYSLOG_SUDO) grep cpu /var/log/syslog
```

Comment:
I didn't mention this before, so you had no way of knowing, but our travis process for the examples is in flux. We still have some things to work out (on my plate, but i'd like to work with you on it when we both have time). Anyway, to keep travis weirdness out of these Makefiles, I think we should just set the above variable via `GET_SVC_OUTPUT ?= ...` so it can be overridden from the environment, then in our .travis.yml (in the top directory), we can override GET_SVC_OUTPUT with sudo at the beginning of the cmd. That way, here you can remove lines 18-23.